src/napari_threedee/manipulators/layer_manipulator.py

```diff
@@ -12,33 +13,57 @@ class LayerManipulator(BaseManipulator):
 
     """
 
-    def __init__(self, viewer, layer, translator_length=50, order=0):
-        self._line_length = translator_length
-        self._initial_translate = None
+    def __init__(self, viewer, layer=None, translator_length=50, order=0):
+        super().__init__(
+            viewer=viewer,
+            layer=layer,
+            translator_length=translator_length,
+            order=order
+        )
+
+    def set_layers(self, layer: napari.layers.Layer):
+        super().set_layers(layer)
 
-        self._centroid = np.array([0, 0, 0])
+    def _initialize_transform(self):
+        if self.layer is not None:
+            self._translation = np.array([0, 0, 0])
+            self._rot_mat = np.eye(3)
+        else:
+            self._translation = np.array([0, 0, 0])
+            self._rot_mat = np.eye(3)
```

Comment:
these look identical, am I missing something?

Reply:
Ah yeah, good catch! They weren't at first (this is why the manipulator wasn't moving in sync with the mouse) and then I didn't catch it when I changed it. I will fix before merging.

Reply:
Fixed!